ui/app/src/modules/Content/Publish/PublishForm.tsx

```diff
@@ -176,28 +179,31 @@ function PublishForm(props: PublishFormProps) {
     showEmailCheckbox,
     publishingChannels,
     publishingChannelsStatus,
-    getPublishingChannels
+    onPublishingChannelsFailRetry,
+    setSubmitDisabled
   } = props;
 
-  useEffect(
-    () => {
-      if (publishingChannels && publishingChannels.length > 0) {
-        setInputs({ ...inputs, 'environment': publishingChannels[0].name });
-      }
-    },
-    // eslint-disable-next-line
-    [publishingChannels]
-  );
-
   const handleInputChange = (name: string) => (e: React.ChangeEvent<HTMLInputElement>) => {
     e.persist();
 
     if (e.target.type === 'checkbox') {
       setInputs({ ...inputs, [name]: e.target.checked });
-    } else if (e.target.type === 'radio' || e.target.type === 'textarea') {
-      setInputs({ ...inputs, [name]: e.target.value })
-    }
+    } else if (e.target.type === 'textarea') {
+      setInputs({ ...inputs, [name]: e.target.value });
+    } else if (e.target.type === 'radio') {
+      const inputValue = e.target.value;
+      setInputs({ ...inputs, [name]: inputValue });
 
+      if (inputValue === 'now') {
+        setTimeout(() => {
```

Comment:
Why is the timeout needed?

Reply:
There was a comment to reset the form values when changing from `Later` to `Now` (after 2 seconds)

Reply:
Timeout should be cancelled if the radio goes back to now